src/axom/sidre/tests/sidre_group.cpp

```diff
@@ -3216,6 +3217,295 @@ TEST(sidre_group, import_conduit_lists)
+  pullTestVals(n3);
+
+  EXPECT_EQ(num_groups, num_groups_chk);
+  EXPECT_EQ(num_views, num_views_chk);
+  EXPECT_EQ(num_views_empty, num_views_empty_chk);
+  EXPECT_EQ(num_views_buffer, num_views_buffer_chk);
+  EXPECT_EQ(num_views_external, num_views_external_chk);
+  EXPECT_EQ(num_views_scalar, num_views_scalar_chk);
+  EXPECT_EQ(num_views_string, num_views_string_chk);
+  EXPECT_EQ(num_bytes_assoc_with_views, num_bytes_assoc_with_views_chk);
+  EXPECT_EQ(num_bytes_external, num_bytes_external_chk);
+  EXPECT_EQ(num_bytes_in_buffers, num_bytes_in_buffers_chk);
+
+  //
+  // Check 4: Recursive check from root Group. Count entire Group hierarchy
+  //
+
+  resetChkVals();
+  num_groups_chk += 1;  // count root Group
+
+  // "A" Group and Views
+  num_groups_chk += 1;        // "A" Group
+  num_views_chk += 3;         // "dat_A1", "dat_A2", "ext_A3" Views
+  num_views_buffer_chk += 2;  // "dat_A1", "dat_A2" Views
+  num_bytes_assoc_with_views_chk += view_A1->getTotalBytes();
+  num_bytes_assoc_with_views_chk += view_A2->getTotalBytes();
+  num_views_external_chk += 1;  // "ext_A3" View
+  num_bytes_external_chk += view_A3->getTotalBytes();
+  num_bytes_in_buffers_chk += view_A1->getTotalBytes();
+  num_bytes_in_buffers_chk += buff2->getTotalBytes();  // "dat_A2" View
+
+  // "B" Group and Views
+  num_groups_chk += 1;
+  num_views_chk += 2;         // "dat_B1", "dat_B2" Views
+  num_views_buffer_chk += 2;  // "dat_B1", "dat_B2" Views
+  num_bytes_assoc_with_views_chk += view_B1->getTotalBytes();
+  num_bytes_assoc_with_views_chk += view_B2->getTotalBytes();
+  num_bytes_in_buffers_chk += view_B1->getTotalBytes();
+  // Note: we do not count view_B2 Buffer bytes since it shares a Buffer
+  //       with view_A2 -- only count Buffer bytes once!!
+
+  // "C" Group and Views
+  num_groups_chk += 1;
+  num_views_chk += 2;         // "val_C1", "val_C2" Views
+  num_views_scalar_chk += 1;  // "val_C1" View
+  num_views_string_chk += 1;  // "val_C2" View
+  num_bytes_assoc_with_views_chk += view_C1->getTotalBytes();
+  num_bytes_assoc_with_views_chk += view_C2->getTotalBytes();
+
+  // "D" Group and Views
+  num_groups_chk += 1;
+  num_views_chk += 2;           // "ext_D1", "dat_D2" Views
+  num_views_external_chk += 1;  // "ext_D1" View
+  num_views_empty_chk += 1;     // "dat_D2" View
+  num_bytes_external_chk += view_D1->getTotalBytes();
+
+  conduit::Node n4;
+  root->getDataInfo(n4);
+
+  pullTestVals(n4);
+
+  EXPECT_EQ(num_groups, num_groups_chk);
+  EXPECT_EQ(num_views, num_views_chk);
+  EXPECT_EQ(num_views_empty, num_views_empty_chk);
+  EXPECT_EQ(num_views_buffer, num_views_buffer_chk);
+  EXPECT_EQ(num_views_external, num_views_external_chk);
+  EXPECT_EQ(num_views_scalar, num_views_scalar_chk);
+  EXPECT_EQ(num_views_string, num_views_string_chk);
+  EXPECT_EQ(num_bytes_assoc_with_views, num_bytes_assoc_with_views_chk);
+  EXPECT_EQ(num_bytes_external, num_bytes_external_chk);
+  EXPECT_EQ(num_bytes_in_buffers, num_bytes_in_buffers_chk);
+
```

Comment:
Should we also test the case where we delete groups, views and/or buffers to ensure that the results match our expectations?

Specifically, it might be instructive to have a test that deletes a group without deleting its data (via `destroyGroup()`) and another that removes the data (via `destroyGroupAndData()`)

Reply:
More test cases added.

Reply:
Thanks!